### PR TITLE
Changed error message E0408 to new format

### PR DESCRIPTION
**Pull request by faebser**
Followed your text and was able to change the ouput to the new format.
I did not encounter any broken test therefore this is a really small commit.

Thanks for letting me hack on the compiler :)

r? @jonathandturner

### CODE REVIEW

**Comment by faebser** [comment]
Part of #35233
Fixes #35698

**Comment by GuillaumeGomez** [comment]
Thanks for your PR!

You also need to update the corresponding test(s) to reflect your changes. They are located in `src/test/compile_fail/`.

**Comment by jonathandturner** [comment]
@faebser - as @GuillaumeGomez mentions, the test infrastructure has some quirks.  One is that the labels aren't tested until you add your first NOTE.  You can read about this in the Extra Credit part of the [blog post](http://www.jonathanturner.org/2016/08/helping-out-with-rust-errors.html)

**Comment by faebser** [comment]
@jonathandturner thanks for the heads up. I will work on the test as soon I have time.

**Comment by GuillaumeGomez** [comment]
@faebser: Don't worry, we're not in a hurry. Just comment here once done.

**Comment by faebser** [comment]
@jonathandturner I don't really get where I have to add the note?
If i try to add a note to https://github.com/rust-lang/rust/blob/master/src/test/compile-fail/E0408.rs (which I think is the original test case that you refer to in your blog post) I just get an unexpected note error if i run the test.

**Comment by jonathandturner** [comment]
@faebser - can you add your changes to this PR so we can take a look?

**Comment by faebser** [comment]
@jonathandturner I added my try as a commit to the PR.
Running ```python src/bootstrap/bootstrap.py --step check-cfail --stage 1``` gives the following errors:

```rust
failures:

---- [compile-fail] compile-fail/E0408.rs stdout ----
	
error: /home/faebser/workspace/rust/src/test/compile-fail/E0408.rs:15: unexpected "error": '15:19: 15:23: variable `y` from pattern #1 is not bound in pattern #2 [E0408]'

error: /home/faebser/workspace/rust/src/test/compile-fail/E0408.rs:15: unexpected "note": 'pattern doesn't bind `y`'

error: /home/faebser/workspace/rust/src/test/compile-fail/E0408.rs:14: expected error not found: variable `y` from pattern #2 is not bound in pattern #1

error: /home/faebser/workspace/rust/src/test/compile-fail/E0408.rs:14: expected note not found: pattern doesn't bind `y`

error: 2 unexpected errors found, 2 expected errors not found
status: exit code: 101
```

**Comment by GuillaumeGomez** [comment]
Then fix them? I don't know what to add actually. :-/

**Comment by faebser** [comment]
@GuillaumeGomez: I am sorry but I am a bit lost and I don't really understand how to fix this error. I added them to this PR because @jonathandturner asked me to.

**Comment by GuillaumeGomez** [comment]
The "errors" you see come from `src/test/compile-fail/E0408.rs`. So, you added annotations (like `//~ ERROR E0408' but some of them aren't matched with the current rustc output (that you updated).

So let's see the first one:

```
E0408.rs:15: unexpected "error": '15:19: 15:23: variable `y` from pattern #1 is not bound in pattern #2 [E0408]'
```

It says that an annotation was not found. So let's look to your file:

```
//~^ ERROR variable `y` from pattern #2 is not bound in pattern #1
```

It means you expect an error to be thrown by the line before this one (the "^" says it). Better like this?

**Comment by faebser** [comment]
@GuillaumeGomez: thanks for your comment. I also realised that **#1** and **#2** were the wrong way around thus generating the error.
I added one more test and will try to find the remaining ones tomorrow.

File: src/test/compile-fail/E0408.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ fn main() {
     let x = Some(0);
 
     match x {
-        Some(y) | None => {} //~ ERROR E0408
-        _ => ()
+        Some(y) | None => {} //~ ERROR variable `y` from pattern #1 is not bound in pattern #2
+        _ => () //~^ NOTE pattern doesn't bind `y`
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/E0408.rs]: Generally, we use "//~|" and not "//~^" when it's on the same line. I mean that the note is thrown by the same line than the error, so it's the "same" issue. With this way to write it, we have the feeling that it's two different errors.

Also, can you align "//~^ NOTE pattern doesn't bind `y`" with "//~ ERROR variable `y` from pattern #1 is not bound in pattern #2" please?

File: src/test/compile-fail/issue-2848.rs [hunk 2]
```diff
@@ -20,6 +20,6 @@ fn main() {
     use bar::foo::{alpha, charlie};
     match alpha {
       alpha | beta => {} //~ ERROR variable `beta` from pattern #2 is not bound in pattern #1
-      charlie => {}
+      charlie => {} //~^ NOTE pattern doesn't bind `beta`
```
Comment (GuillaumeGomez) [inline src/test/compile-fail/issue-2848.rs]: Same as above.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
Once these fixed, please squash your commits. If you don't know how to do it, take a look [here](https://blog.guillaume-gomez.fr/Git-tips/1/2).

**Comment by faebser** [comment]
@GuillaumeGomez can you please review the third test too? thanks

**Comment by GuillaumeGomez** [comment]
All good for me. Please squash your commits now.

**Comment by KiChjang** [comment]
This looks like it only requires squashing, what's blocking this for 2 weeks?

**Comment by GuillaumeGomez** [comment]
No clue. :-/

**Comment by faebser** [comment]
Sorry for the delay, I thought that the squash was okay but after having another look today I saw that it did not work. I am working on fixing it right now.

**Comment by faebser** [comment]
@GuillaumeGomez: squash is okay now

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**